### PR TITLE
PXP-638: [CLI] Implement new config

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary

Build the workflow for the wukong init command to satisfy the specifications. Details about the command inputs and outputs can be found [here](https://mindvalley.atlassian.net/wiki/spaces/PXP/pages/762773518/RFC17+-+New+Global+Init+UX).

Ticket: https://mindvalley.atlassian.net/browse/PXP-524

## What's Changed

- [x] Update the Google auth to store in our config file instead of a separate file
- [x] Update all auth logics to refer to new path
- [x] Add google cloud and vault login to `wukong init`
- [x] Updated wukong login logic to use same function on `wukong init` to avoid duplicate codes

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/183#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Fadhil — APPROVED** [review]
Ok. This looks good to go. We're mostly refactoring some config vars to use the new format with an `auth` section. Let's test it out.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/183#pullrequestreview-1817448334" title="Approved">Fadhil</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/183#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/config.rs
Comment (jk-gan) [inline cli/src/config.rs:100]: is it `vault` or `bunker`? because it is `[auth.bunker]` in the RFC

File: cli/Cargo.toml
Comment (jk-gan) [inline cli/Cargo.toml:66]: I'm not sure if we should use `anyhow` for error handling or not, since we already have our own Error Code

File: cli/src/auth/google_cloud.rs
Comment (jk-gan) [inline cli/src/auth/google_cloud.rs:90]: any reason why you want to use this format? okta and vault are using RFC3339
- https://github.com/mindvalley/wukong-cli/blob/main/cli/src/auth/okta.rs#L235
- https://github.com/mindvalley/wukong-cli/blob/main/cli/src/auth/vault.rs#L231

**Review by jk-gan — CHANGES_REQUESTED** [review]

File: cli/src/auth/google_cloud.rs
Comment (mfauzaan) [reply cli/src/auth/google_cloud.rs:90]: Oh wait, i thought both okta and vault are using ISO8601, but i'll change this to RFC339

**Comment by jk-gan** [comment]
As I know, `wukong init` command should re-initialize the config file. How about the UX in this scenario:
The user had already logged in google cloud and bunker, but they choose to not log in when they run `wukong init`. Should we remove the google cloud and bunker credentials from the newly generated config file?

@onimsha @mfauzaan

File: cli/src/config.rs
Comment (mfauzaan) [reply cli/src/config.rs:100]: Need to confirm this with @onimsha 
Yeah all of our code refers to vault, but new RFC says bunker. if we need to use new term we should update all the codes to avoid any confusion

File: cli/src/commands/init.rs
Comment (jk-gan) [inline cli/src/commands/init.rs:80]: But the default value for google login is `false`, unless you have a reason why they are different, I think we should make them consistent, either both `true` or both `false`.

File: cli/src/commands/init.rs
Comment (jk-gan) [inline cli/src/commands/init.rs:63]: The default value for bunker auth is `true`.

**Review by jk-gan — CHANGES_REQUESTED** [review]

File: cli/Cargo.toml
Comment (mfauzaan) [reply cli/Cargo.toml:66]: Yeah, i tried couple implemenations to avoid using anyhow, but this [line](https://github.com/mindvalley/wukong-cli/blob/d5972682b525c45529ba4c2676faa2aa4b16d412/cli/src/auth/google_cloud.rs#L81) forces me to use the anyhow.

I couldn't make it working with our handler, but if you have any idea how to archive this yeah we can remove this as we are doing our own handling

Comment (jk-gan) [reply cli/Cargo.toml:66]: ahhh, my bad. I missed this line. Since this is a trait method, then we have to use live with it.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/183#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/183#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/183#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/183#pullrequestreview-1837748231" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
